### PR TITLE
Fix 'Local files not deleted' when using bad 'skip_file' entry

### PR DESCRIPTION
**Pull request by abraunegg**
* Add a check for bad 'skip_file' entry that prevents incorrect searching of local changes (mainly deletes) from being synced.

### CODE REVIEW

**Comment by abraunegg** [comment]
#383 and #479 were both caused by config options being used where:
```
Config option 'skip_file'           = .*|~*
```

This patch checks for the offending item `.*` and forces the application to exit if detected:

**without patch**
See #383 and #479

**with patch**
```
(dmd-2.085.0)[alex@centos7full fix-bad-config-no-local-delete]$ ./onedrive --confdir '~/.config/onedrive-personal' --synchronize --verbose 
Using Config Dir: /home/alex/.config/onedrive-personal
Initializing the OneDrive API ...
Opening the item database ...
All operations will be performed in: /home/alex/OneDrivePersonal
ERROR: Invalid skip_file entry '.*' detected
(dmd-2.085.0)[alex@centos7full fix-bad-config-no-local-delete]$ 
```

File: src/main.d [hunk 1]
```diff
@@ -282,8 +282,17 @@ int main(string[] args)
 	log.vdebug("skip_dir: ", cfg.getValueString("skip_dir"));
 	selectiveSync.setDirMask(cfg.getValueString("skip_dir"));
 	log.vdebug("Configuring skip_file ...");
-	log.vdebug("skip_file: ", cfg.getValueString("skip_file"));
-	selectiveSync.setFileMask(cfg.getValueString("skip_file"));
+	// Validate skip_file to ensure that this does not contain an invalid configuration
+	// Do not use a skip_file entry of .* as this will prevent correct searching of local changes to process.
+	if (canFind(cfg.getValueString("skip_file"), ".*")) {
```
Comment (norbusan) [inline src/main.d]: No, that is not good, because it also exempts settings like `temp.*` and similar.

Comment (norbusan) [reply src/main.d]: With `skip_file = "abc.*"` I still get `ERROR: Invalid skip_file entry '.*' detected`

**Comment by abraunegg** [comment]
**default config**
```
(dmd-2.085.0)[alex@centos7full fix-bad-config-no-local-delete]$ ./onedrive --confdir '~/.config/onedrive-personal' --display-config
onedrive version                    = v2.3.3-3-gdaec620
Config path                         = /home/alex/.config/onedrive-personal
Config file found in config path    = true
Config option 'check_nosync'        = false
Config option 'sync_dir'            = /home/alex/OneDrivePersonal
Config option 'skip_dir'            = 
Config option 'skip_file'           = ~*|.~*|*.tmp
Config option 'skip_dotfiles'       = false
Config option 'skip_symlinks'       = false
Config option 'monitor_interval'    = 45
Config option 'min_notify_changes'  = 5
Config option 'log_dir'             = /var/log/onedrive/
Selective sync configured           = false
(dmd-2.085.0)[alex@centos7full fix-bad-config-no-local-delete]$ ./onedrive --confdir '~/.config/onedrive-personal' --synchronize --verbose 
Using Config Dir: /home/alex/.config/onedrive-personal
Initializing the OneDrive API ...
Opening the item database ...
All operations will be performed in: /home/alex/OneDrivePersonal
Initializing the Synchronization Engine ...
Account Type: personal
Default Drive ID: 66d53be8a5056eca
Default Root ID: 66D53BE8A5056ECA!101
Remaining Free Space: 5368709120
Fetching details for OneDrive Root
OneDrive Root exists in the database
Syncing changes from OneDrive ...
Applying changes of Path ID: 66D53BE8A5056ECA!101
Applying changes of Path ID: BC7D88EC1F539DCF!1288
Uploading differences of .
Processing root
...
Uploading new items of .
Applying changes of Path ID: 66D53BE8A5056ECA!101
Applying changes of Path ID: BC7D88EC1F539DCF!1288
(dmd-2.085.0)[alex@centos7full fix-bad-config-no-local-delete]$ 
```

**invalid configuration**
```
(dmd-2.085.0)[alex@centos7full fix-bad-config-no-local-delete]$ ./onedrive --confdir '~/.config/onedrive-personal' --display-config        
onedrive version                    = v2.3.3-3-gdaec620
Config path                         = /home/alex/.config/onedrive-personal
Config file found in config path    = true
Config option 'check_nosync'        = false
Config option 'sync_dir'            = /home/alex/OneDrivePersonal
Config option 'skip_dir'            = 
Config option 'skip_file'           = .*|~*
Config option 'skip_dotfiles'       = false
Config option 'skip_symlinks'       = false
Config option 'monitor_interval'    = 45
Config option 'min_notify_changes'  = 5
Config option 'log_dir'             = /var/log/onedrive/
Selective sync configured           = false
(dmd-2.085.0)[alex@centos7full fix-bad-config-no-local-delete]$ ./onedrive --confdir '~/.config/onedrive-personal' --synchronize --verbose 
Using Config Dir: /home/alex/.config/onedrive-personal
Initializing the OneDrive API ...
Opening the item database ...
All operations will be performed in: /home/alex/OneDrivePersonal
ERROR: Invalid skip_file entry '.*' detected
(dmd-2.085.0)[alex@centos7full fix-bad-config-no-local-delete]$ 
```

**valid configuration but containing invalid character sequence**
```
(dmd-2.085.0)[alex@centos7full fix-bad-config-no-local-delete]$ ./onedrive --confdir '~/.config/onedrive-personal' --display-config        
onedrive version                    = v2.3.3-3-gdaec620
Config path                         = /home/alex/.config/onedrive-personal
Config file found in config path    = true
Config option 'check_nosync'        = false
Config option 'sync_dir'            = /home/alex/OneDrivePersonal
Config option 'skip_dir'            = 
Config option 'skip_file'           = temp.*|~*
Config option 'skip_dotfiles'       = false
Config option 'skip_symlinks'       = false
Config option 'monitor_interval'    = 45
Config option 'min_notify_changes'  = 5
Config option 'log_dir'             = /var/log/onedrive/
Selective sync configured           = false
(dmd-2.085.0)[alex@centos7full fix-bad-config-no-local-delete]$ ./onedrive --confdir '~/.config/onedrive-personal' --synchronize --verbose 
Using Config Dir: /home/alex/.config/onedrive-personal
Initializing the OneDrive API ...
Opening the item database ...
All operations will be performed in: /home/alex/OneDrivePersonal
Initializing the Synchronization Engine ...
Account Type: personal
Default Drive ID: 66d53be8a5056eca
Default Root ID: 66D53BE8A5056ECA!101
Remaining Free Space: 5368709120
Fetching details for OneDrive Root
OneDrive Root exists in the database
Syncing changes from OneDrive ...
Applying changes of Path ID: 66D53BE8A5056ECA!101
Applying changes of Path ID: BC7D88EC1F539DCF!1288
Uploading differences of .
Processing root
...
Uploading new items of .
Applying changes of Path ID: 66D53BE8A5056ECA!101
Applying changes of Path ID: BC7D88EC1F539DCF!1288
(dmd-2.085.0)[alex@centos7full fix-bad-config-no-local-delete]$ 
```

**Comment by norbusan** [comment]
Thanks, that worked

**Pull request merged**